Add spec for governance token staking from general account and locked rewards counting towards staking balance (#2350)
vegaprotocol/specs: protocol/0057-TRAN-transfers.md

Allowing users to initiate transfers allows for the following capabilities:

- A user can transfer funds from a public key A to a public key B.
- A user can transfer governance tokens from general account to another key's general account.

> - A user can transfer governance tokens from and to a locked account used for staking [LOCKED_FOR_STAKING](./0059-STKG-simple_staking_and_delegating.md).

Reviewer: Don't think we need this any more? Tokens should contribute automatically without transfer.

Allowing users to initiate transfers allows for the following capabilities:

- A user can transfer funds from a public key A to a public key B.

> - A user can transfer funds from and to a locked account used for staking [LOCKED_FOR_STAKING](./0059-STKG-simple_staking_and_delegating.md).

Reviewer 2: ```suggestion
- A user can transfer the governance token only from and to a locked account used for staking [LOCKED_FOR_STAKING](./0059-STKG-simple_staking_and_delegating.md).
```

Allowing users to initiate transfers allows for the following capabilities:

- A user can transfer funds from a public key A to a public key B.

> - A user can transfer the governance token only from and to a locked account used for staking [LOCKED_FOR_STAKING](./0059-STKG-simple_staking_and_delegating.md).

Reviewer 2: This isn't true. You can also transfer vega tokens from general account to another key's general account.

- As a user I can transfer funds from a general account I control to an other party's general account. Such transfer can be immediate or delayed. (<a name="0057-TRAN-001" href="#0057-TRAN-001">0057-TRAN-001</a>)
- As a user I **cannot** transfer funds from a general account I control to reward account with a one-off transfer. (<a name="0057-TRAN-002" href="#0057-TRAN-002">0057-TRAN-002</a>)

> - As a user I can transfer funds from a general account I control to a LOCKED_FOR_STAKING account. Such transfer can be immediate or delayed. (<a name="0057-TRAN-003" href="#0057-TRAN-003">0057-TRAN-003</a>).

Reviewer 2: ```suggestion
- As a user I can transfer the governance token from a general account I control to a LOCKED_FOR_STAKING account. Such transfer must be immediate. (<a name="0057-TRAN-003" href="#0057-TRAN-003">0057-TRAN-003</a>).
```

> - As a user I can transfer funds from a LOCKED_FOR_STAKING account under my control to any party's general account. Such transfer can be immediate or delayed.  (<a name="0057-TRAN-004" href="#0057-TRAN-004">0057-TRAN-004</a>)

Reviewer 2: ```suggestion
- As a user I can transfer the governance token from a LOCKED_FOR_STAKING account under my control to any party's general account. Such transfer must be immediate.  (<a name="0057-TRAN-004" href="#0057-TRAN-004">0057-TRAN-004</a>)
```

- As a user I can transfer funds from a general account I control to an other party's general account. Such transfer can be immediate or delayed. (<a name="0057-TRAN-001" href="#0057-TRAN-001">0057-TRAN-001</a>)
- As a user I **cannot** transfer funds from a general account I control to reward account with a one-off transfer. (<a name="0057-TRAN-002" href="#0057-TRAN-002">0057-TRAN-002</a>)
- As a user I can transfer the governance token from a general account I control to a LOCKED_FOR_STAKING account. Such transfer must be immediate. (<a name="0057-TRAN-003" href="#0057-TRAN-003">0057-TRAN-003</a>).

> - As a user I can transfer the governance token from a LOCKED_FOR_STAKING account under my control to any party's general account. Such transfer must be immediate.  (<a name="0057-TRAN-004" href="#0057-TRAN-004">0057-TRAN-004</a>)

Reviewer: Think these need to be added to features.json as well.
